lib/src/Base/Stat/ProcessSampleImplementation.cxx

```diff
@@ -350,6 +350,34 @@ ProcessSampleImplementation ProcessSampleImplementation::computeQuantilePerCompo
   return result;
 }
 
+/*
+ * Method computeQuantilePerComponent() gives the quantile per component of the sample using Robbins Monro approximation
+ */
+Field ProcessSampleImplementation::computeQuantilePerComponentRobbinsMonro(const Scalar prob) const
+{
+  const UnsignedInteger size = getSize();
+  if (size == 0) return Field();
+  if (size == 1) return Field(mesh_, data_[0]);
+  Scalar exponent = ResourceMap::GetAsScalar("ProcessSample-RobbinsMonroExponent");
+  if (!(exponent > 0.5 && exponent <= 1.0)) exponent = 0.75;
+  const UnsignedInteger dimension = data_[0].getDimension();
+  const UnsignedInteger length = data_[0].getSize();
+  Point quantiles(data_[0].getImplementation()->getData());
+  Point scale(data_[1].getImplementation()->getData() - quantiles);
+  for (UnsignedInteger j = 0; j < scale.getSize(); ++j)
+    scale[j] = std::abs(scale[j]);
+  for (UnsignedInteger i = 1; i < size; ++i)
+    {
+      const Point & Xi = data_[i].getImplementation()->getData();
+      const Scalar tau = 1.0 / std::pow(i, exponent);
+      for (UnsignedInteger j = 0; j < scale.getSize(); ++j)
+	quantiles[j] -= scale[j] * tau * (int(Xi[j] <= quantiles[j]) - prob);
```

Comment:
Why did you introduce scale?  It is not mentioned in the paper AFAICT, and results look better without it.

Reply:
The theoretical convergence is conditioned to the convergence of a series related to the decreasing deterministic sequence of step sizes. It is unchanged if a scale is introduced (ie we have an almost sure convergence if scale!=1 iff there is an almost sure convergence if scale==1), but in practice it can change things a lot. If you take scale=1, the update on the quantile estimator is either step_i or step_i * (1-q) depending on X_i being greater or non greater than the current quantile estimate, with step_i = 1/i^alpha, 0.5<alpha<=1. Suppose that your distribution is widely spread or significantly non centered. The quantile will be large and it will be obtained by an accumulation of tiny steps if scale=1. With scale=X_1-X_0, there is a hope to capture the actual spread of the distribution and to accelerate the estimator.
It is more a heuristic than a mathematical result, but it sounds reasonable to me. You (and anybody else) are welcome to confirm/disproof this idea ;-)
Here is a script to check that things are going the right way:
```
import openturns as ot
from time import time

m = 1000
n = 50000
ot.Log.Show(ot.Log.INFO)
t0 = time()
print("generate data")
scale = 0.1
dt = 1.0 / (m - 1.0)
mesh = ot.RegularGrid(0.0, dt, m)
amplitude = 1.0
mean = 10.0
trend = ot.TrendTransform(ot.SymbolicFunction('t', str(mean)), mesh)
p = ot.GaussianProcess(trend, ot.SquaredExponential([scale], [amplitude]), mesh)
s = p.getSample(n)
print("t=", time() - t0, "s")

# Quantiles
q = 0.5
theoretical = ot.Normal(mean, amplitude).computeQuantile(q)[0]
t0 = time()
print("empirical quantile")
empirical = s.computeQuantilePerComponent(q)
print("t=", time() - t0, "s")
t0 = time()
print("Robbins Monro quantile")
robbins_monro = s.computeQuantilePerComponentRobbinsMonro(q)
print("t=", time() - t0, "s")
# Check the distribution of empirical quantiles and robbins_monro quantiles
indep_bw = ot.KernelSmoothing().computeMixedBandwidth(empirical.getValues())
# Adjust bandwidth to take into account dependence
bw = indep_bw * (scale / dt)**(1.0/5.0)
graph = ot.KernelSmoothing().build(empirical.getValues(), bw).drawPDF()
graph.add(ot.KernelSmoothing().build(robbins_monro.getValues(), bw).drawPDF())
yMax = graph.getBoundingBox().getUpperBound()[1]
graph.add(ot.Curve([[theoretical, 0.0], [theoretical, yMax]]))
graph.setColors(["blue", "green", "red"])
graph.setLegends(["empirical", "Rob.Mon.", "theor."])
ot.Show(graph)
```

Reply:
With scale:
![robbins-scale](https://user-images.githubusercontent.com/161011/42367873-d1162e56-8105-11e8-946a-80507b562da0.png)

Without scale:
![robbins-woscale](https://user-images.githubusercontent.com/161011/42367890-dc0cdba2-8105-11e8-89c7-0c437499c0bf.png)

Reply:
Please check also with q=0.05 or q=0.95.

Reply:
I added
```
diff = empirical.getValues() - robbins_monro.getValues()
print(diff.getMin(), diff.getMax(), diff.computeMean(), diff.computeStandardDeviation())
```
at the end of your script.
With scale:
```
[-0.852258] [1.09544] [0.0132888] [[ 0.128232 ]]
```
Without scale:
```
[-0.011142] [0.01822] [0.00314498] [[ 0.00544802 ]]
```